### PR TITLE
MdeModulePkg/VariableStandaloneMm: Fix TCG MOR secure feature issue

### PR DESCRIPTION
**Pull request by xuweiintel**
# Description

- In VariableStandaloneMm, the VariableHaveTcgProtocols() always returns FALSE, it causes TCG MOR secure feature does not work in standalone MM environment.

- The commit fda8f631edbbf3823760542a06f12bd60fd39181 added the check agaist Tcg/Tcg2 protocols to detele and lock OS-created MOR variable to fix the issue that Fedora 24 and 25 follow the TCG's Platform Reset Attack Mitigation spec and unintentionally create MOR variable.

As Fedora 24 and 25 are EOL today, in order to fix the MOR feature issue in standalone MM environment, revert the above commit fda8f631edbbf3823760542a06f12bd60fd39181.

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

1. Boot to windows
2. Shutdown by pluging out the power.
3. Power on and boot.
4. TcgMor is processed.

## Integration Instructions

N/A

### CODE REVIEW

File: MdeModulePkg/Universal/Variable/RuntimeDxe/VariableStandaloneMm.c [hunk 1]
```diff
@@ -87,21 +87,3 @@ VariableServiceInitialize (
...
-                protocol database
-  @retval FALSE Neither the TCG nor the TCG2 protocol is installed in the UEFI
-                protocol database
-**/
-BOOLEAN
-VariableHaveTcgProtocols (
-  VOID
-  )
-{
-  return FALSE;
```
Comment (ardbiesheuvel) [inline MdeModulePkg/Universal/Variable/RuntimeDxe/VariableStandaloneMm.c:106]: I have no objections to the PR as-is. However, given that the only purpose of `VariableHaveTcgProtocols()` appears to be this check, and the conditional update of the variable, we could also just return TRUE here, no? We might want to rename this function to make it more obvious what the narrow use-case is for which it exists.

Comment (xuweiintel) [reply MdeModulePkg/Universal/Variable/RuntimeDxe/VariableStandaloneMm.c:106]: Thanks a lot for the review. Sorry, I'm not sure if I understand correctly. 
Do you mean we can keep all the code in [TcgMorLockSmm.c], but only rename VariableHaveTcgProtocols() and make it return TRUE in [VariableStandaloneMm.c]?
What name do you suggest to rename VariableHaveTcgProtocols() to?

Comment (xuweiintel) [reply MdeModulePkg/Universal/Variable/RuntimeDxe/VariableStandaloneMm.c:106]: > I have no objections to the PR as-is. However, given that the only purpose of `VariableHaveTcgProtocols()` appears to be this check, and the conditional update of the variable, we could also just return TRUE here, no? We might want to rename this function to make it more obvious what the narrow use-case is for which it exists.

Hi @ardbiesheuvel,
Thanks a lot for review.
Do you mean we can keep all the code in [TcgMorLockSmm.c], but only rename VariableHaveTcgProtocols() and make it return TRUE in [VariableStandaloneMm.c]?

**Review by ardbiesheuvel — APPROVED** [review]

File: MdeModulePkg/Universal/Variable/RuntimeDxe/VariableStandaloneMm.c (hunk 1 above)
Comment (niruiyu) [reply MdeModulePkg/Universal/Variable/RuntimeDxe/VariableStandaloneMm.c:106]: Based on the discussion in https://edk2.groups.io/g/devel/topic/106795434, I guess it's safe just to revert the original commit.

**Review by niruiyu — APPROVED** [review]

**Comment by mdkinney** [comment]
@mergify rebase

**Pull request merged**